src/backend/distributed/shared_library_init.c

```diff
@@ -435,6 +450,23 @@ RegisterConnectionCleanup(void)
 }
 
 
+/*
+ * RegisterClientBackendCounterDecrement is called when the backend terminates.
+ * For all client backends, we register a callback that will undo
+ */
+static void
+RegisterClientBackendCounterDecrement(void)
+{
+	static bool registeredCleanup = false;
+	if (registeredCleanup == false)
```

Comment:
!registeredCleanup

also, could use an extra newline since this is a very different situation from the usual assign-and-check.